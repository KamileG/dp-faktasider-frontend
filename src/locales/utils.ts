import { useContext } from 'react';
import { LocaleContext } from './LocaleContext';

export type SupportedLanguage = 'en' | 'nn' | 'nb';

export const useLocale = () => {
  return useContext(LocaleContext).lang;
};
