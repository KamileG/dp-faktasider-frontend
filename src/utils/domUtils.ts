export function getDomElemnt(id: string) {
  return document.getElementById(id);
}

export function pxFromTop(id: string): number {
  return getDomElemnt(id)?.getBoundingClientRect().top || Number.POSITIVE_INFINITY;
}
