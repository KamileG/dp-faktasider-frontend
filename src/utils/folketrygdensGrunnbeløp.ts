export function useGrunnbellop() {
  const G = 99858;

  return {
    GtoNOK: (g: number) => Math.round(g * G).toLocaleString('nb-NO'),
    G: G,
  };
}
