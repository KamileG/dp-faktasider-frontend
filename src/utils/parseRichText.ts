export type SanityBlock = {
  _key?: string;
  _type?: string;
  sanityChildren?: SanitySpan[];
  style?: string;
  list?: string;
};

export type SanitySpan = {
  _key?: string;
  _type?: string;
  marks?: string[];
  text?: string;
};

export function parseBlocksTilBolker(blocks: SanityBlock[]): SanityBlock[] {
  let newBlocks: SanityBlock[] = [];
  let currentBolk: SanityBlock | undefined = undefined;

  blocks.forEach((block) => {
    if (block.style === 'h2') {
      currentBolk = {
        _type: 'bolk',
        sanityChildren: [block],
      };
      newBlocks.push(currentBolk);
    } else if (currentBolk) {
      currentBolk.sanityChildren.push(block);
    } else {
      newBlocks.push(block);
    }
  });

  return newBlocks;
}

function parseRichText(richText) {
  return parseBlocksTilBolker(richText);
}

export default parseRichText;
