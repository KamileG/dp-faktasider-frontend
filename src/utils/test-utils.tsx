import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { TranslationsProvider } from '../i18n/i18nextConfig';
import { VisForContextProvider } from '../components/BlockContent/VisFor/VisForContext';
import { DevContextProvider } from '../components/DevKnapper/DevContext';
import { LocaleProvider } from '../i18n/LocaleContext';

// https://testing-library.com/docs/react-testing-library/setup#custom-render
const AllTheProviders = ({ children }) => (
  <LocaleProvider lang="no">
    <VisForContextProvider>
      <DevContextProvider>
        <TranslationsProvider>{children}</TranslationsProvider>
      </DevContextProvider>
    </VisForContextProvider>
  </LocaleProvider>
);

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, 'queries'>) =>
  // @ts-ignore
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
