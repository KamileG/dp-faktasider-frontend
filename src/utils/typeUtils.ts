// https://stackoverflow.com/questions/41285211/overriding-interface-property-type-defined-in-typescript-d-ts-file
export type Modify<T, R> = Omit<T, keyof R> & R;
