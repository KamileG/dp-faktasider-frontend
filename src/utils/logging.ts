/*
Vi må lazy-loade loggingConfig, da gatsby pre-rendrer html, og kresjer dermed amplitude-sdk
Prøver derfor å ikke initiere amplitude-sdk instansen før en logevent blir kalt.
*/

import { isProduction } from './environment';

const loggEvent = (event: string, ekstraData?: object) => {
  if (!isProduction()) {
    return;
  }

  import('../utils/loggingConfig').then((logging) =>
    logging.loggInstance.logEvent(event, {
      ...ekstraData,
      appName: 'dp-faktasider',
    })
  );
};

export const loggError = (error: Error, ekstraData?: object) =>
  loggEvent('Error', {
    ...ekstraData,
    siteUrl: window.location.pathname,
    msg: error.message,
    name: error.name,
    stack: error.stack,
  });

export const loggRedirect = (fraLenke: string) => loggEvent('Redirect fra gammel lenke', { fraLenke });

export const loggSidevisning = (side: string) => loggEvent('Side vist', { side });

export const loggIkkeOversatt = (side: string) => loggEvent('Ikke oversatt', { side });

export const loggKalkulatorbruk = (type: string) => loggEvent('Brukt kalkulator', { type });

export const loggHashlenkeKlikk = () => loggEvent('Brukt hashlenke');

export const loggMeny = (type: string) => loggEvent('Meny', { type });

export const loggH2MenyKlikk = () => loggEvent('H2Meny', { type: 'Hopp til underoverskrift' });

export const loggVisTilleggsinfo = (tittel: string) => loggEvent('Vis tilleggsinfo', { tittel });

export const loggTilpassInnhold = (situasjon: string) => loggEvent('Tilpasset innhold', { situasjon });

export const loggNotFound = (path: string) => loggEvent('404 - not found', { path });

export const loggHashNotFound = (hash: string, path: string) => loggEvent('Hashlenke finnes ikke', { path, hash });
