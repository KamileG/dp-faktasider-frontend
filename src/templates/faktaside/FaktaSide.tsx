import * as React from 'react';
import { graphql, PageProps } from 'gatsby';
import BlockContent from '../../components/BlockContent/BlockContent';
import Layout from './Layout';
import GraphQLErrorList from '../../components/GraphqlErrorList';
import parseRichText from '../../utils/richTextUtils/parseRichText';
import ErrorBoundary from '../../components/ErrorBoundary';
import { SanityBlock } from '../../utils/richTextUtils/richTextTypes';
import { getBolkTitler } from '../../utils/richTextUtils/getBolkTitler';
import localizeSanityContent from '../../i18n/localizeSanityContent';
import { FaktasideProvider } from './FaktasideContext';
import { SupportedLanguage } from '../../i18n/supportedLanguages';
import SEO from '../../components/SEO';
import IkkeOversatt from './IkkeOversatt';
import { SistOppdatert } from './SistOppdatert';
import RelatertInformasjon from './RelatertInformasjon';
import { Translations } from '../../types/translations';

export const query = graphql`
  query FaktaSide($id: String) {
    side: sanityFaktaSide(id: { eq: $id }) {
      _rawTitle
      _rawInnhold
      _rawIngress
      _rawSistOppdatert
      _rawRelatertInformasjon
      _rawPublisert
      slug {
        current
      }
    }
  }
`;

interface PageContext {
  lang: SupportedLanguage;
  id: string;
}

export interface FaktaSideData {
  side: {
    _rawTitle?: string;
    _rawInnhold?: SanityBlock[];
    _rawIngress?: string;
    _rawSistOppdatert?: string;
    _rawRelatertInformasjon?: SanityBlock[];
    _rawPublisert?: Translations<boolean>;
    slug: {
      current: string;
    };
  };
}

export interface FaktaSideProps extends PageProps<FaktaSideData, PageContext> {
  errors: any;
}

function FaktaSide(props: FaktaSideProps) {
  const lang = props.pageContext.lang;
  const data = localizeSanityContent(props.data, lang) as FaktaSideData;

  const erPublisert = props.data.side._rawPublisert?.[lang];
  if (!erPublisert) {
    return <IkkeOversatt {...props} />;
  }

  const parsedRichText = parseRichText(data.side._rawInnhold);
  const bolkTitler = getBolkTitler(parsedRichText, data.side._rawRelatertInformasjon);

  const tittel = data.side._rawTitle || '';
  const description = data.side._rawIngress || '';

  return (
    <ErrorBoundary>
      <SEO title={tittel} description={description} lang={lang} />
      <FaktasideProvider faktasideProps={props}>
        <Layout header={tittel} menuItems={bolkTitler} ingress={description}>
          <GraphQLErrorList errors={props.errors} />
          <SistOppdatert>{data.side._rawSistOppdatert}</SistOppdatert>
          <BlockContent blocks={parsedRichText} />
          <RelatertInformasjon blocks={data.side._rawRelatertInformasjon} />
        </Layout>
      </FaktasideProvider>
    </ErrorBoundary>
  );
}

export default FaktaSide;
