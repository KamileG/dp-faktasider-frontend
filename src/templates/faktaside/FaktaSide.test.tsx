import React from 'react';
import { render } from '../../utils/test-utils';
import FaktaSide from './FaktaSide';
import { createFaktasideContext } from '../../../gatsby-utils/createFaktasider';
import { rawFaktasideDataMock } from '../../../__mocks__/rawFaktasideDataMock';

jest.mock('../../utils/faktasiderSummary/useFaktasiderSumary', () => () => {
  return require('../../utils/faktasiderSummary/faktasiderSummaryMockData').faktasiderSummaryMockData;
});

beforeAll(() => {
  const JSutils = require('nav-frontend-js-utils');
  JSutils.guid = jest.fn(() => 'Helt tilfeldig ID');
});

/* Dette er en veldig grov test som fanger opp det meste av endringer pÃ¥ faktaside */
test('Snapshottest Faktaside', () => {
  const context = createFaktasideContext(rawFaktasideDataMock, 'no');
  const { container } = render(
    // @ts-ignore
    <FaktaSide pageContext={context} />
  );

  expect(container).toMatchSnapshot();
});
