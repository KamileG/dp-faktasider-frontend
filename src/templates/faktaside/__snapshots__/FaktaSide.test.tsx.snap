// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshottest Faktaside 1`] = `
.c20 {
  position: relative;
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  pointer-events: none;
}

.c22 {
  position: absolute;
  top: -2.5rem;
}

.c22:focus {
  box-shadow: none;
}

.c21 {
  -webkit-transition: 0.1s;
  transition: 0.1s;
}

.c21:focus-within {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #254b6d;
  height: 100%;
  width: 100%;
  border-radius: 0.2rem;
}

.c26 {
  position: absolute;
  right: calc(-1rem - 0.6em);
  -webkit-transform: translateY(0.1em);
  -ms-transform: translateY(0.1em);
  transform: translateY(0.1em);
  color: #0067C5;
}

.c24 {
  position: relative;
}

.c27 {
  stroke: currentColor;
  fill: none;
  stroke-linecap: round;
  stroke-linejoin: round;
  stroke-width: 2.5;
  height: calc(0.5rem + 0.4em);
  opacity: 0.6;
  -webkit-transition: 0.15s;
  transition: 0.15s;
}

.c25:hover .c27 {
  opacity: 1;
}

.c19 {
  background-color: white;
  padding: 1.5rem 3rem;
  border-radius: .2rem;
  margin-bottom: 2rem;
  max-width: 100vw;
  position: relative;
}

.c23 {
  background-color: #fffe;
  padding: 1rem;
  text-align: center;
  position: -webkit-sticky !important;
  position: sticky !important;
  z-index: 10;
  margin-bottom: 1.5rem;
}

.c8 {
  background-color: white;
  padding: 1.5rem 0 2.5rem;
}

.c9 {
  display: block;
  font-size: 1.1rem;
  font-weight: 600;
  padding: 0.5rem 1.5rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c9:hover {
  background-color: #ddd8;
}

.c13 {
  background: none;
  color: inherit;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  outline: inherit;
  border-radius: 50%;
  height: 2.2rem;
  width: 2.2rem;
  opacity: 0.9;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c13:hover {
  opacity: 0.7;
}

.c13:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #254b6d;
}

.c15 {
  width: 50%;
  stroke: black;
  stroke-width: 0.4;
  stroke-linecap: round;
}

.c12 {
  position: absolute;
  top: 0;
  right: 0;
  -webkit-transition: top 0.2s;
  transition: top 0.2s;
  z-index: 100;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  border-radius: 0.5rem;
  -webkit-filter: drop-shadow(0 0.5rem 0.5rem #0005);
  filter: drop-shadow(0 0.5rem 0.5rem #0005);
  pointer-events: auto;
}

.c11 {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  pointer-events: none;
}

.c16 {
  -webkit-transition: max-height 0.2s;
  transition: max-height 0.2s;
  max-height: calc(100vh - 0px);
  max-width: 25rem;
  overflow-y: auto;
  border-radius: 0.5rem;
  display: none;
}

.c14 {
  margin: 1rem;
  background-color: white;
  position: absolute;
  right: 0;
  top: 0;
  box-shadow: 0 0 0 0.2rem #8886;
}

.c7 {
  position: -webkit-sticky;
  position: sticky;
  max-height: calc(100vh - 0px);
  overflow-y: auto;
  -webkit-transition: top 0.2s,max-height 0.2s;
  transition: top 0.2s,max-height 0.2s;
  max-width: 16rem;
}

.c2 {
  background-color: #CCE1F3;
  border-bottom: 0.3rem solid #99C2E8;
  border-top: 0.3rem solid #99C2E8;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
}

.c4 {
  font-size: 1.7rem;
}

.c3 {
  padding: 2rem;
  max-width: 50rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #E9E7E7;
  border-bottom: 0.1rem solid #bbb;
  padding: 1rem;
}

.c1 {
  width: 14.375rem;
  margin: 0;
}

.c1 label {
  position: absolute !important;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px 1px 1px 1px);
  clip: rect(1px 1px 1px 1px);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c17 {
  -webkit-flex: 0 1 40rem;
  -ms-flex: 0 1 40rem;
  flex: 0 1 40rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: auto;
  max-width: 40rem;
  padding: 1rem 0 4rem;
}

.c17 ul {
  padding-left: 2rem;
  list-style: disc;
  margin: 1.5rem 0;
}

.c17 ul li {
  margin-bottom: 1rem;
  line-height: 1.3;
}

.c17 ul ul {
  list-style: circle;
}

.c17 ol {
  list-style: decimal;
  padding-left: 1.5rem;
  margin-bottom: 1.5rem;
}

.c17 ol li {
  margin: 0.5rem 0;
  line-height: 1.3;
  padding-left: 1rem;
}

.c17 p {
  margin: 0.5rem 0 1rem;
}

.c6 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  position: relative;
  min-height: 30vh;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c18 {
  padding: 1rem;
  text-align: center;
}

.c28 {
  text-align: center;
}

.c28 ul {
  list-style: none;
}

@media (max-width:65rem) {
  .c19 {
    padding: 1.5rem 5vw;
  }
}

@supports (position:-webkit-sticky) or (position:sticky) {
  .c12 {
    position: -webkit-sticky;
    position: sticky;
    top: 0px;
  }
}

@media not all and (max-width:65rem) {
  .c11 {
    display: none;
  }
}

@media (max-width:65rem) {
  .c7 {
    display: none;
  }
}

@media not all and (max-width:65rem) {
  .c10 {
    display: none;
  }
}

@media (min-width:65rem) {
  .c17 {
    padding: 0 2rem;
  }
}

@media (min-width:65rem) {
  .c6 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
  }

  .c6 .order-1 {
    -webkit-order: 1;
    -ms-flex-order: 1;
    order: 1;
  }

  .c6 .order-2 {
    -webkit-order: 2;
    -ms-flex-order: 2;
    order: 2;
  }

  .c6 .order-3 {
    -webkit-order: 3;
    -ms-flex-order: 3;
    order: 3;
  }
}

@media (min-width:65rem) {
  .c28 ul {
    -webkit-column-count: 2;
    column-count: 2;
  }

  .c28 ul a {
    white-space: nowrap;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="skjemaelement c1"
    >
      <label
        class="skjemaelement__label"
        for="Helt tilfeldig ID"
      >
        Velg språk
      </label>
      <div
        class="selectContainer input--fullbredde"
      >
        <select
          aria-invalid="false"
          class="skjemaelement__input"
          id="Helt tilfeldig ID"
        >
          <option
            disabled=""
            value=""
          >
            Velg språk
          </option>
          <option
            value="no"
          >
            Norsk
          </option>
          <option
            value="en"
          >
            English
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <h1
        class="typo-sidetittel c4"
      >
        Dette er litt testdata
      </h1>
      <p
        class="typo-normal"
      >
        Denne brukes i test
      </p>
    </div>
  </div>
  <div
    class="c5"
  >
    <div
      class="c6"
    >
      <nav
        aria-labelledby="Helt tilfeldig ID"
        class="c7 order-1"
        style="top: 0px;"
      >
        <h2
          class="sr-only"
          id="Helt tilfeldig ID"
        >
          Sideoversikt
        </h2>
        <ol
          class="c8"
        >
          <li>
            <a
              class="c9 lenke"
              href="/no/arbeidsledig/"
            >
              Arbeidsledig
               
              
            </a>
          </li>
          <li>
            <a
              class="c9 lenke"
              href="/no/permittert/"
            >
              Permittert
               
              (no)
            </a>
          </li>
        </ol>
      </nav>
      <nav
        aria-labelledby="Helt tilfeldig ID"
        class="c10 order-1"
      >
        <h2
          class="sr-only"
          id="Helt tilfeldig ID"
        >
          Sideoversikt
        </h2>
        <div
          class="c11"
        >
          <div
            class="c12"
          >
            <button
              aria-expanded="false"
              class="c13 c14"
            >
              <svg
                class="c15"
                viewBox="0.5 0.5 3 3"
              >
                <circle
                  cx="1"
                  cy="1"
                  r=".12"
                />
                <path
                  d="M 1.8 1 L 3.3 1"
                />
                <circle
                  cx="1"
                  cy="2"
                  r=".12"
                />
                <path
                  d="M 1.8 2 L 3.3 2"
                />
                <circle
                  cx="1"
                  cy="3"
                  r=".12"
                />
                <path
                  d="M 1.8 3 L 3.3 3"
                />
              </svg>
              <span
                class="sr-only"
              >
                Sideoversikt
              </span>
            </button>
            <div
              class="c16"
            >
              <ol
                class="c8"
              >
                <li>
                  <a
                    class="c9 lenke"
                    href="/no/arbeidsledig/"
                  >
                    Arbeidsledig
                     
                    
                  </a>
                </li>
                <li>
                  <a
                    class="c9 lenke"
                    href="/no/permittert/"
                  >
                    Permittert
                     
                    (no)
                  </a>
                </li>
              </ol>
            </div>
          </div>
        </div>
      </nav>
      <main
        class="c17 order-2"
      >
        <div
          class="c18"
        >
          Denne siden ble sist oppdatert 10. juli 2020 12:54
        </div>
        <article
          aria-labelledby="Helt tilfeldig ID"
          class="c19"
          data-testid="overskrift"
          style="position: relative;"
        >
          <div
            class="c20"
          >
            <div
              class="c21"
            >
              <div
                class="c22"
                id="overskrift"
                tabindex="-1"
              />
            </div>
          </div>
          <h2
            class="c23 group-header-hash-link-hide"
            id="Helt tilfeldig ID"
            style="top: 0px;"
          >
            Overskrift
            <span
              aria-hidden="true"
              class="c24"
            >
              <a
                class="c25 c26"
                href="#overskrift"
              >
                <svg
                  class="c27"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
                  />
                  <path
                    d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
                  />
                </svg>
              </a>
            </span>
          </h2>
          <div>
            <p>
              Litt innhold
            </p>
            <ul>
              <li>
                Punkt 1
              </li>
              <li>
                Punkt 2
              </li>
            </ul>
          </div>
        </article>
        <div
          class="c28"
        >
          <article
            aria-labelledby="Helt tilfeldig ID"
            class="c19"
            data-testid="relatert-informasjon"
            style="position: relative;"
          >
            <div
              class="c20"
            >
              <div
                class="c21"
              >
                <div
                  class="c22"
                  id="relatert-informasjon"
                  tabindex="-1"
                />
              </div>
            </div>
            <h2
              class="c23 group-header-hash-link-hide"
              id="Helt tilfeldig ID"
              style="top: 0px;"
            >
              Relatert informasjon
              <span
                aria-hidden="true"
                class="c24"
              >
                <a
                  class="c25 c26"
                  href="#relatert-informasjon"
                >
                  <svg
                    class="c27"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
                    />
                    <path
                      d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
                    />
                  </svg>
                </a>
              </span>
            </h2>
            <p>
              Relatert info
            </p>
          </article>
        </div>
      </main>
    </div>
  </div>
</div>
`;
